### PR TITLE
Dynamic skill menu

### PR DESCRIPTION
**Pull request by 3akev**

### CODE REVIEW

File: src/Tilesets/EffectsTileset.tscn [hunk 1]
```diff
@@ -1,9 +1,3 @@
-[gd_scene load_steps=2 format=2]
-
-[ext_resource path="res://Assets/effects/energy_shield.png" type="Texture" id=1]
+[gd_scene format=2]
```
Comment (nightblade9) [inline src/Tilesets/EffectsTileset.tscn:1]: Do we even need this tileset any more?

File: src/Ecs/Systems/SkillMenuSystem.gd [hunk 2]
```diff
@@ -0,0 +1,30 @@
+extends "res://Ecs/Core/System.gd"
+
+var _event_bus
+var _skillmenu
```
Comment (nightblade9) [inline src/Ecs/Systems/SkillMenuSystem.gd:4]: Should this be `_skill_menu`?

File: src/Ecs/Systems/SkillMenuSystem.gd [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+	_event_bus.connect("set_player", self, "on_set_player")
+
+func on_set_player(player):
+	if player.has("SkillComponent") and OS.has_touchscreen_ui_hint():
+		clear_skillmenu()
+		var skills = player.get("SkillComponent").skills
+		for skill in skills:
+			var index = skills.find(skill)
+			var skill_button = create_skill_button(skill, index)
+			skill_button.position.x = 64 * index
```
Comment (nightblade9) [inline src/Ecs/Systems/SkillMenuSystem.gd:18]: This is a big assumption. A strategy I used in the past is to keep a map such as:

`const SKILL_INDEX_X = {
  "bow": 64,
  "energy shield": 128
}`

Then you can reference it as `skill_button.position.x = SKILL_INDEX_X[skill]` or something.

**Review by nightblade9 — APPROVED** [review]

File: src/Ecs/Systems/SkillMenuSystem.gd (hunk 3 above)
Comment (3akev) [reply src/Ecs/Systems/SkillMenuSystem.gd:18]: > This is a big assumption

I'm sorry, what is? We're multiplying by the index, so they're ordered as they are in `SkillComponent`. Fixing them with a map isn't as generalised.

File: src/Tilesets/EffectsTileset.tscn (hunk 1 above)
Comment (3akev) [reply src/Tilesets/EffectsTileset.tscn:1]: Yeah. Wouldn't make sense to use the humans tileset in the effects tilemap.

**Pull request merged**